File: 2374-node-with-highest-edge-score/2374-node-with-highest-edge-score.cpp
class Solution {
public:
    int edgeScore(vector<int>& edges) {
        double n = edges.size(), maxScore = 0, ansNode = n + 1;
        vector<double> score(n);
        for(int i = 0; i < n; ++i) {
            score[edges[i]] += i;
            maxScore = max(maxScore, score[edges[i]]);
        }
        for(int i = 0; i < n; ++i) {
            if(score[i] == maxScore) {
                ansNode = i;
                break;
            }
        }
        return (int)ansNode;
    }
};

File: 2374-node-with-highest-edge-score/2374-node-with-highest-edge-score.py
class Solution:
    def edgeScore(self, edges: List[int]) -> int:
        n, maxScore, ansNode = len(edges), 0, -1
        score = [0] * (n)
        for i in range (n):
            score[edges[i]] += i
            maxScore = max(maxScore, score[edges[i]])
        for i in range (n):
            if score[i] == maxScore:
                ansNode = i
                break
        return ansNode